skyportal/schema.py

```diff
@@ -76,21 +77,28 @@ def setup_schema():
                     "Model class ends with 'Schema'"
                 )
 
-            class Meta(object):
-                model = class_
-                sqla_session = _DBSession
+            def add_schema(schema_class_name, exclude=[]):
+                schema_class_meta = type(f'{schema_class_name}_meta', (),
+                                         {'model': class_, 'sqla_session': _DBSession,
+                                          'ordered': True, 'exclude': []}
+                )
+                for exclude_attr in exclude:
+                    if hasattr(class_, exclude_attr) and getattr(class_, exclude_attr) is not None:
+                        schema_class_meta.exclude.append(exclude_attr)
+
+                schema_class = type(schema_class_name, (_ModelSchema,),
+                                    {'Meta': schema_class_meta}
+                )
 
-            schema_class_name = '%s' % class_.__name__
+                if 'id' not in exclude:
```

Comment:
Overall, this now looks great!

This if-statement is a bit magical.  Perhaps a keyword like `install_schema` or `add_to_model` would make things more explicit.

Then, let's add a docstring to explain this to our future selves, and I think we're done!

Reply:
Good call on just adding that as a parameter. That's in now, as is the docstring.